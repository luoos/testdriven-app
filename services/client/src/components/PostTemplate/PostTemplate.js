import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const PostTemplate = (props) => {
  return (
    <>
      <div className="content">
        {props.title_clickable ?
          <h1><Link to={`/post/${props.id}`}>{props.title}</Link></h1>
          :
          <h1>{props.title}</h1>
        }
        <p className="subtitle is-4">{new Date(props.created_time*1000).toDateString()}</p>
        <p className="content">{props.content}</p>
      </div>
      <br/>
    </>
  );
}

PostTemplate.defaultProps = {
  title_clickable: true
}

PostTemplate.prototype = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  created_time: PropTypes.number.isRequired,
  last_updated_time: PropTypes.number
}

export default PostTemplate;