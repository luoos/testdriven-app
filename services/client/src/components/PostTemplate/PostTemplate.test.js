import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import PostTemplate from './PostTemplate';

const postData = {
  id: 1,
  title: 'a big name',
  content: 'what do you want to eat for lunch',
  created_time: new Date('2019-02-20'),
  last_updated_time: new Date('2019-02-22')
}

describe('<PostTemplate />', () => {

  it('PostTemplate renders properly title clickable', () => {
    const wrapper = shallow(<PostTemplate {...postData} />)
    const title = wrapper.find('Link');
    expect(title.length).toBe(1);
    expect(title.get(0).props.children).toBe(postData.title);
    const content = wrapper.find('p.content');
    expect(content.length).toBe(1);
    expect(content.get(0).props.children).toBe(postData.content);
  });

  it('PostTemplate renders properly title unclickable', () => {
    const wrapper = shallow(<PostTemplate {...postData} title_clickable={false} />)
    const title = wrapper.find('h1');
    expect(title.length).toBe(1);
    expect(title.get(0).props.children).toBe(postData.title);
    const content = wrapper.find('p.content');
    expect(content.length).toBe(1);
    expect(content.get(0).props.children).toBe(postData.content);
  });

  it('renders a snapshot properly', () => {
    const tree = renderer.create(<MemoryRouter location="/"><PostTemplate {...postData} /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
})