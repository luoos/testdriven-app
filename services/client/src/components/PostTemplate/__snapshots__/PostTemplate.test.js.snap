// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTemplate /> renders a snapshot properly 1`] = `
Array [
  <div
    className="content"
  >
    <h1>
      <a
        href="/post/1"
        onClick={[Function]}
      >
        a big name
      </a>
    </h1>
    <p
      className="subtitle is-4"
    >
      Fri Mar 22 51107
    </p>
    <p
      className="content"
    >
      what do you want to eat for lunch
    </p>
  </div>,
  <br />,
]
`;
